File: app/src/main/java/space/levan/memory/di/module/ActivityModule.java
package space.levan.memory.di.module;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;
import space.levan.memory.di.scope.ActivityScope;

/**
 * Activity Module
 *
 * @author WangZhiYao
 * @date 2017/10/17
 */

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        this.mActivity = activity;
    }

    @Provides
    @ActivityScope
    public Activity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/space/levan/memory/di/module/ActivityModule.kt
package space.levan.memory.di.module

import android.support.v7.app.AppCompatActivity
import dagger.Module
import dagger.Provides
import space.levan.memory.di.scope.ActivityScope

/**
 * File description
 *
 * @author WangZhiYao
 * @date 2018/1/15
 */

@Module
class ActivityModule(private val activity: AppCompatActivity) {

    @Provides
    @ActivityScope
    fun provideActivity(): AppCompatActivity {
        return activity
    }
}